### ISSUE TITLE
Too long httpheaders generate http 500 response, not 400 (bad request)

### ISSUE THREAD

**Issue by Erwinvandervalk** [Bobris/Nowin#34]
I've found an issue with Nowin. Somehow I got into a situation where too many cookies were being added. When that happened, I got a http 500 response from nowin. After some debugging, i've narrowed the line of code where it occurs:

https://github.com/Bobris/Nowin/blob/master/Nowin/Transport2HttpHandler.cs#L346 

It looks like nowin get's lost with parsing the http header and then throws an NotImplementedException. 

The request header length was about 9000 bytes. So it's clearly too long, but if nowin would have returned HTTP 400 - Bad Request (Request Header too long), then it would have been immediately clear what's wrong.

If you want, I can create a repro in the form of a unit test..

Here's the http header I used:
``` httprequest
GET http://localhost:1080 / HTTP/1.1
Host: localhost:1080
Connection: keep-alive
Pragma: no-cache
Cache-Control: no-cache
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8
User-Agent: Mozilla/5.0 (Windows NT 6.3; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/43.0.2357.124 Safari/537.36
Accept-Encoding: gzip, deflate, sdch
Accept-Language: en-US,en;q=0.8,nl;q=0.6
Cookie: SignInMessage.9f1cc9c4ed409eb9ace9fa453d19aae0=AQAAANCMnd8BFdERjHoAwE_Cl-sBAAAAa4ZDCDenmUyKVkQePQm0tQAAAAACAAAAAAADZgAAwAAAABAAAADIXgTFFJyVWBBgictppYfnAAAAAASAAACgAAAAEAAAALeN6HgMMnouwbgBQXEV6FcIAwAA4VJaAW-qi3u6BCxvquXiod0I_QeAjpd80a1EQFvFuVuvTaVbbns6fMgaCSIIBCHl2nBBGZslErptFB9o71tRt0aTXdEnHIlAcRvzg5wqp-8BhlyGjF2rnXqsjaYeNnWfZrUC_FxQEiLwAYMD0kulYduplAGM71_eqzdwKd4eNzmJnB0rV5bBia5iwHZM3o71gz1omZuGPE7A91LMLfX8463o38bnTzfxdar724fORJ7hhbn2TFW-sNBXxVMWOYuQT7LVfCyFZQiu7m_AFWTpM9IMX60U9GWWgNO8RgRpZQShyHZU2Cso71rghIHJXQY4aifmTcn4bwzQ8rPo_vffqn-myCuUccIu5PjzudC2He37VDMZU12NwLXnmMAvM4ipfFrTaRuVEblpKO9gsq-qGiflyHF5cu09hvZmykN_snkipeM_RdI48bwD-S_jy1OpLo2wOYUtebho_lfWHgVU_muezYsw1CF2K93CX5cBJH4j7wkerrnZwHPk75Q03O6wzLYf-2NlD1NZI4U1Ny_jwRXkVpqWulyrMrjwJxXzFBqtH80WHvsmTpd3-vAvQhRgMWUv_HOqezN8YQ5HzuWPcMO8sDHIpcn_gKRjXf4CUuM37YbXQ33CvRksLyuIBfDZtn-Biwg6HFjlLVoJ1Mkdqwh60Ibc_h8TbvI6WfC69VR3DHy14M-K5UbU0bJm4ar0b3Luzgd9W4-_4gSNFWgUoosvs9bXEKS3oxI-anMOFByx0Z-_w5yGA8d2jrOGT_ojOiRCXSIhUTGYqnNddMJtuLxI1auK-M0QqPfvqsUGkKM74-J1hLWogkN2Lx_up4hv77k496kgm5qkveLGUjYcUy10rKo_q6C7-WVVJNlpKPPg1xOT_QiaLUuNX3ziQ3JGAcspvSXjCy5rKpXGmQf5PqthtkE245sI90ncAI8F479v5SSDAvBH_id2PuvyGrFGIJL4q1YCU5WIs1Y6tS0LQOYiki4bGZmvJaq6s4NWoIld6fEggo94ZVEUHUYpm9dMIzPMeO1u-QcUAAAAPNSe-TP2AgqpqMzHLXVUyNetcow; OpenIdConnect.nonce.7oEtF53WxOi2uAwJ%2B4wawzdGZbb3Cxb3gQoOipf%2F%2Feg%3D=QVFBQUFOQ01uZDhCRmRFUmpIb0F3RV9DbC1zQkFBQUFhNFpEQ0Rlbm1VeUtWa1FlUFFtMHRRQUFBQUFDQUFBQUFBQURaZ0FBd0FBQUFCQUFBQUJCZ1RaZm1ybkRYM0JnR2NuOWo0ZWpBQUFBQUFTQUFBQ2dBQUFBRUFBQUFFaHlwZ0hPVFl0aEp1RTBwRXM4YmdHQUFBQUFtVEFVRll6SUxBTDI4UUxjMnVEY0tmeG9hSk5EX0l5Nk1GRXFSbjl5ZWdsSmRXYTR5M1FwQzktTThMYWVCRDlzTGZ5WGFqQjJBWHNOM3lkT3JCUExxc3NyZEJXWWZpcDhMMVFnQXNlVmJoRWZ2X2RKY0VWZUZ5LWpWTjJZOEZrWkJhajFjWXVUZUJYSmJmai1BREJTRl8xUG9FUUpMY1NkRDZNYThVMVJXRE1VQUFBQUVDMzEyLUFVbHQ1eXNERVN3cE83Mm1BX1U2SQ%3D%3D; OpenIdConnect.nonce.b3Ccpijv%2F67GFS37gwl5rSPNSHVQ%2B8ZziQfKjG67eOo%3D=QVFBQUFOQ01uZDhCRmRFUmpIb0F3RV9DbC1zQkFBQUFhNFpEQ0Rlbm1VeUtWa1FlUFFtMHRRQUFBQUFDQUFBQUFBQURaZ0FBd0FBQUFCQUFBQUNiZUNpSWFfWHdZbWd5ak1NNE5CMmZBQUFBQUFTQUFBQ2dBQUFBRUFBQUFPTl9qNkRaZjJmSGdEbVExTjlJZkxDQUFBQUF4Q3EwQXZiQVktd1lLX0pQYVprR2V2aFpOMnNhRWE1MTNtSndfekJnUkJtOXViZXRNZ1I1OU9yWjZPM0pIc3VhQ25CTl9hRVJBSklVUF9PX29sS0V3a240Q3l4eTlnbXJlNzRCVmM4TGZIUFVYUXhSTmpnLThFekgxZG1LMEc2d0w4R0d2TXlqS1BxRHJpMFgtUFFrRW84dXppdTVtNk1BRHZKalFtb1BMVFFVQUFBQWhHQ1MxYWFROVBQaS1aNmQyZjB4aHAwQnpPRQ%3D%3D; OpenIdConnect.nonce.JtVEDvgzzDmfUti0Uy8h0R4AwMSCp2Y4eNLeEVxARu0%3D=QVFBQUFOQ01uZDhCRmRFUmpIb0F3RV9DbC1zQkFBQUFhNFpEQ0Rlbm1VeUtWa1FlUFFtMHRRQUFBQUFDQUFBQUFBQURaZ0FBd0FBQUFCQUFBQUJScFN6bUpUTVRRNXpvb2d4VEdxUHhBQUFBQUFTQUFBQ2dBQUFBRUFBQUFOYW1rYkpIZXhaYl9mVVNpVWlqYTB5QUFBQUFyd0gzYXBhcGtOU09IYkpicEFzN3dfaEp0dzQ4MDJIVG9NTmEwcDBUY284OFpQNU9OVEhDeWNObkVTaGVkQ1JGTkNiR2U3Wi0xc096MG8wdDhjaDFOeWh6NG9YaTcyTUdCUXpuaHJNVWZYa2VpZU5lcDVwck10VXRQbkZjUk92U09IbVFCTDFSM0E5MVMxTlU0TzlrTUNYUXRmWThlT0xmekN1aVVJMDZmSXdVQUFBQWY4WGlrTWFHTkRoYXg5dm5RaGRCMVE4bnlJVQ%3D%3D; OpenIdConnect.nonce.E1aZE0raJlAnQ8F9ub8QIgXdwCoVUOWNjjNnZCTeNYw%3D=QVFBQUFOQ01uZDhCRmRFUmpIb0F3RV9DbC1zQkFBQUFhNFpEQ0Rlbm1VeUtWa1FlUFFtMHRRQUFBQUFDQUFBQUFBQURaZ0FBd0FBQUFCQUFBQURmN2tBdkFLMnpvUkQ0OVVNQVdwTmRBQUFBQUFTQUFBQ2dBQUFBRUFBQUFFdVZOMFNKSDdwMUVGS2dxaDFDc1dDQUFBQUFXZHhXdkRPSlktRXgzTHFVR1V0MjV0NXJuZHFnNWJLdTJ0N1VFVEU5X3FXbHdxckVLdWFZTjVEcFVKbkFva3lpeU5mOFlKTklpcFlNekhUQmVVNVYwRm82ZmlBeG5NVUtVRkxMa2gtdUkwVHVnLWRTT3VhY2RCbS01MXdtbGVIV0R1QmM4aHlxRlhOTGpUQ1JIb3Rxb3NCaHlwRk0wWk5mOTk3TG5DZmNpYXdVQUFBQWRvY0VZVk1wWmJRZURKM29lWV9vSjZuVG8wYw%3D%3D; OpenIdConnect.nonce.ZF9aPs0xeqsCxWd5mw0Br881uyrJoESZh%2BMBXRlCnuc%3D=QVFBQUFOQ01uZDhCRmRFUmpIb0F3RV9DbC1zQkFBQUFhNFpEQ0Rlbm1VeUtWa1FlUFFtMHRRQUFBQUFDQUFBQUFBQURaZ0FBd0FBQUFCQUFBQUF0UUhBaXcxTmJRemRueW9hRkpjZTJBQUFBQUFTQUFBQ2dBQUFBRUFBQUFGeE5mQ3Babi03cWJZM0NRMGxoY3FtQUFBQUFuTFRXN3RaRFVqTklOc05jUjRmNWhtcjRiV3l3QUNVblF2bkFnSGpfUGozYm5RLWFhWmVnN3FlRlNnUXBwZEc4czRHYmlVUHhqVUt4dWFkZThPVWI3aWthaGkzWFdxbUFyeks1NmJPYUdiZ1RQRjlMbFhmT0NRNDFoa1RmQ3RZTW5qNnh4d09lZ0FnVkw4UzlYU0RBbDZkUzlVcmdVYnJWdkF2eTl1dmtYQ1FVQUFBQS1pdFcwTVhzVGt5TW4ydVgxWm5aNE1FbWNBcw%3D%3D; OpenIdConnect.nonce.GayIEQf25qrkIina1vkxUfaDQpFLuopElvnkOVSiNps%3D=QVFBQUFOQ01uZDhCRmRFUmpIb0F3RV9DbC1zQkFBQUFhNFpEQ0Rlbm1VeUtWa1FlUFFtMHRRQUFBQUFDQUFBQUFBQURaZ0FBd0FBQUFCQUFBQUFFZVQxREZCa2lfX0ZjNWJ1QTBTanRBQUFBQUFTQUFBQ2dBQUFBRUFBQUFJMUV6V3NEVUJDMlJValFrWXVyeXBTQUFBQUF3RXlEUFFaQnlOTzlEQ0w1U25LelNVdjJCV0toaWhIazFnc1pGZHVQdTNfT3Jia21pTjdnVVgtSGg0WDhRVDJzeGlxVXRHVUFCRXBNMHZxbEkxSjVBdXJrUnNsUHF1UldkYjhReGpBREgtV1pBZFFHUGgxTEFHZGowcDFfekprY0N5S1Z6RUUtUEVzM0xVZVN5NVdvUldVcFlIazEzTTlocmRSb3dydFo2VXNVQUFBQVpLWjZSTlBubnFjUDVoZmZzRHF5ZGc4WUprdw%3D%3D; OpenIdConnect.nonce.LpTcgsQ6SoVMYXTXXWE5LLw%2B8A7sF%2FiM18Dpgj9NMIc%3D=QVFBQUFOQ01uZDhCRmRFUmpIb0F3RV9DbC1zQkFBQUFhNFpEQ0Rlbm1VeUtWa1FlUFFtMHRRQUFBQUFDQUFBQUFBQURaZ0FBd0FBQUFCQUFBQUFOMDMwUDVqRktlLXFkYVN0YXVkZ3VBQUFBQUFTQUFBQ2dBQUFBRUFBQUFDbVA2ZWJ0M29UX3FHa2dVTzJZcTRTQUFBQUFJcHBYd01Nd3dZTWcyTUgyTjNJWUtuZWdzRG5pUnpPVUpvVFg5S1hCM3dxWWotaHlVNHlzZTRpZ1RLU2YyNlFIeXpmdVdYREpLN0MwZ0VUUGtYNnVLelR2Ti1Cdjg5ZTRvQU9KcEN2bklUaUtmRjRZWTEwSlFaOGcwdGR4ZjVqU0x4OFlXUTl4dlJYNHJZTy1FT2lHel9QMzhvdDViTk1VZTRGbjBpaWdhckFVQUFBQTc4UkJNMDVxeWtqNWRCMG5QUG9iZktFU3VvOA%3D%3D

```

**Comment by Bobris** [comment]
Failing test would be perfect. I am surprised that you think that it is because that line as it is just about parsing Path and your first line is really not valid imho (first line must have only 3 parts delimited by spaces, but you have 4 parts ...)

**Comment by Bobris** [comment]
BTW you can increase this 8192 byte limit by configuration of course it will use more memory...

**Comment by Erwinvandervalk** [comment]
Thanks for your reply!

 

I’ve created a PR with a failing test. https://github.com/Bobris/Nowin/pull/35 

 

Best regards,

Erwin

**Comment by Erwinvandervalk** [comment]
Well.. it might postpone the issue, but the real bug was that somehow we were getting too many cookies. The server failing with a 500 error got me confused a bit though about the cause. 

 

Btw.. really impressive work on Nowin.. thanks for creating it :)

**Comment by Bobris** [comment]
Fixed in 0.17.0
